src/core/inc/GslMatrix.h

```diff
@@ -197,6 +197,17 @@ class GslMatrix : public Matrix
   //! This function multiplies \c this matrix by vector \c x and returns the resulting vector.
   GslVector  multiply                  (const GslVector& x) const;
 
+  //! This function multiplies \c this matrix by vector \c x and fills
+  //the preallocated vector \c y.
+  void       multiply                  (const GslVector& x, GslVector& y) const;
+
+  //! This function multiplies \c this matrix by matrix \c X and returns the resulting matrix.
+  GslMatrix  multiply                  (const GslMatrix& X) const;
```

Comment:
Yeah, this was on my list of "stuff to optimize after getting correct results", but since we don't end up calling this function from within the likelihood, only from within the setup, I'm no longer planning to bother.